scoap3/utils/throttling.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework.settings import api_settings
+from rest_framework.throttling import SimpleRateThrottle
+
+
+class UserGroupThrottle(SimpleRateThrottle):
+    def __init__(self):
+        pass
+
+    def get_rate(self) -> str | None:
+        if not self.user_groups:
+            return api_settings.DEFAULT_THROTTLE_RATES["DefaultUser"]
+        elif "Admin" in self.user_groups:
+            return api_settings.DEFAULT_THROTTLE_RATES["Admin"]
+        elif "Advanced_user" in self.user_groups:
+            return api_settings.DEFAULT_THROTTLE_RATES["Advanced_user"]
+        elif "API_user" in self.user_groups:
+            return api_settings.DEFAULT_THROTTLE_RATES["API_user"]
+
+    def allow_request(self, request, view) -> bool:
+        self.user_groups = request.user.groups.values_list("name", flat=True)
```

Comment:
lets check if `request.user` exists, to be sure

Reply:
It return an empty query set even with anon users